src/App.tsx

```diff
@@ -1,26 +1,365 @@
+              data-cy="ToggleAllButton"
+              onClick={toggleAllTodos}
+            />
+          )}
+
+          {/* Add a todo on form submit */}
+          <form
+            onSubmit={e => {
+              e.preventDefault();
+              const trimmed = newTodoTitle.trim();
+
+              if (trimmed) {
+                handleAddingTodo(trimmed);
+              } else {
+                setErrorMessage('Title should not be empty');
+              }
+            }}
+          >
+            <input
+              ref={inputRef}
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={newTodoTitle}
+              onChange={e => setNewTodoTitle(e.target.value)}
+              disabled={isAddingTodo}
+              autoFocus
+            />
+          </form>
+        </header>
+
+        <TodoList
+          todos={filteredTodos}
+          editingTodoId={editingTodoId}
+          editingTitle={editingTitle}
+          loadingTodoIds={loadingTodoIds}
+          tempTodo={tempTodo}
+          onToggle={(todo: Todo) =>
+            handleToggleCompleted(todo.id, todo.completed)
+          }
+          onDelete={(todo: Todo) => handleDeleteTodo(todo.id)}
+          onEditStart={(todo: Todo) => {
+            setEditingTodoId(todo.id);
+            setEditingTitle(todo.title);
+          }}
+          onEditChange={(value: string) => setEditingTitle(value)}
+          onEditConfirm={(todo: Todo) => handleEditTitle(todo.id)}
+          onEditCancel={() => {
+            setEditingTodoId(null);
+            setEditingTitle('');
+          }}
+        />
+
+        {/* Hide the footer if there are no todos */}
+        {todos.length > 0 && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="TodosCounter">
+              {todos.filter(todo => !todo.completed).length} items left
+            </span>
+
+            {/* Active link should have the 'selected' class */}
+            <nav className="filter" data-cy="Filter">
+              <a
+                href="#/"
+                className={
+                  activeFilter === 'all'
+                    ? 'filter__link selected'
+                    : 'filter__link'
+                }
+                data-cy="FilterLinkAll"
+                onClick={() => setActiveFilter('all')}
+              >
+                All
+              </a>
+
+              <a
+                href="#/active"
+                className={
+                  activeFilter === 'active'
+                    ? 'filter__link selected'
+                    : 'filter__link'
+                }
+                data-cy="FilterLinkActive"
+                onClick={() => setActiveFilter('active')}
+              >
+                Active
```

Comment:
Use `Object.values(your created enum)` and render these options with `map()` method